src/bin/pg_autoctl/keeper_pg_init.c

```diff
@@ -86,6 +87,169 @@ keeper_pg_init_fsm(Keeper *keeper, KeeperConfig *config)
+	if (!configurationPathPresent)
+	{
+		log_error("Cannot find configuration file directory");
+		return false;
+	}
+
+	join_path_components(defaultPosgresqlConfPath, defaultConfigurationPath, "postgresql.conf");
+
+
+	log_error("Moving %s to %s", defaultPosgresqlConfPath, posgresqlConfPath );
+	if (!move_file(defaultPosgresqlConfPath, posgresqlConfPath))
+	{
+		return false;
+	}
+
+	/* edit postgresql.conf to disable settings */
+	read_file(posgresqlConfPath, &fileContents, &fileSize);
+	disable_configuration_parameter_inplace(fileContents, fileSize, "data_directory");
+	disable_configuration_parameter_inplace(fileContents, fileSize, "hba_file");
+	disable_configuration_parameter_inplace(fileContents, fileSize, "ident_file");
+	disable_configuration_parameter_inplace(fileContents, fileSize, "include_dir");
+	write_file(fileContents, fileSize, posgresqlConfPath);
+
+	free(fileContents);
+
+
+	join_path_components(hbaConfPath, pgdata, "pg_hba.conf");
+
+	if(!file_exists(hbaConfPath))
+	{
+
+		log_warn("hba file does not exist : %s, need to copy", hbaConfPath);
+		join_path_components(defaultHbaConfPath, defaultConfigurationPath, "pg_hba.conf");
+
+		move_file(defaultHbaConfPath, hbaConfPath);
+	}
+
+
+	join_path_components(identConfPath, pgdata, "pg_ident.conf");
+
+	if(!file_exists(identConfPath))
+	{
+
+		log_warn("pg_ident.conf file does not exist : %s, need to copy", identConfPath);
+		join_path_components(defaultIdentConfPath, defaultConfigurationPath, "pg_ident.conf");
+
+		move_file(defaultIdentConfPath, identConfPath);
+	}
+
+	/* disable auto start in default configuration */
+	{
+		char defaultStartConfPath[MAXPGPATH];
+		char newStartConfPath[MAXPGPATH];
+		char *newStartConfData = "disabled";
+
+		join_path_components(defaultStartConfPath, defaultConfigurationPath, "start.conf");
+		join_path_components(newStartConfPath, defaultConfigurationPath, "start.conf.orig");
+
+		read_file(defaultStartConfPath, &fileContents, &fileSize);
+		write_file(fileContents, fileSize, newStartConfPath);
+
+		write_file(newStartConfData, strlen(newStartConfData), defaultStartConfPath);
+
+	}
+	return true;
+}
+
+
+/*
+ * disable_configuration_parameter_inplace looks for a pattern
+ * [parameterName] = [parameter value] and puts # in front to disable the
+ * setting. Note that the function would change "port = 5432" to "#port= 5432".
+ * It will remove the space between parameter name and = to make room for #
+ * in the beginning in order to perform modification in place.
+ */
+static void disable_configuration_parameter_inplace(char *fileContents, long fileSize, char *parameterName)
+{
+	int offset = 0;
+	char *substring = NULL;
+	char searchString[MAXPGPATH];
+
+	snprintf(searchString, MAXPGPATH, "%s =", parameterName);
+
+	while ( (substring = strstr(fileContents + offset, searchString)) != NULL)
+	{
+		offset = substring - fileContents;
+		*substring = '#';
+		memcpy(substring+1, parameterName, strlen(parameterName));
+	}
```

Comment:
I don't think that loop read the fileContents line by line and matches with the parameter name only when it's the first thing in the line (ignoring white space). It's pretty hard for me to make sure the function implements what we need here.